### ISSUE TITLE
Question About the First Line of the Documentation

### ISSUE THREAD

**Issue by ZiedMayel** [Husqvik/GraphQlClientGenerator#81]
@Husqvik 
Can you please explain what is thie meaning of this line? 

`GraphQlClientGenerator.Console --serviceUrl <GraphQlServiceUrl> --outputPath <TargetPath> --namespace <TargetNamespace> [--authorization <header value>]`

**Comment by Husqvik** [comment]
The generation can be invoked in various way. The first example is using the console app `GraphQlClientGenerator.Console`. For successful operation the generator has to be fed with necessary input which is:

- GraphQL schema defining all the data types, their fields, operations and other metadata - in the example above this is done by fetching the schema from `GraphQlServiceUrl` URL
- `outputPath` a file name into which all the generated C# members will be generated
- `namespace` - to put the generated client into a namespace that fits your application
- optional parameters, there is more than just headers; if you run the `GraphQlClientGenerator.Console` without parameters it prints out the full list

```
GraphQL C# Client Generator Console App 0.9.0
Copyright 2017-2020

ERROR(S):
  Required option 'o, outputPath' is missing.
  Required option 'n, namespace' is missing.

  -o, --outputPath           Required. Output path; include file name for single file output type; folder name for one class per file output type

  -n, --namespace            Required. Root namespace all classes and other members are generated to

  -u, --serviceUrl           GraphQL service URL used for retrieving schema metadata

  -s, --schemaFileName       Path to schema metadata file in JSON format

  --headers                  Format: {Header}:{Value}; allows to enter custom headers required to fetch GraphQL metadata

  --classPrefix              Class prefix; value "Test" extends class name to "TestTypeName"

  --classSuffix              Class suffix, for instance for version control; value "V2" extends class name to "TypeNameV2"

  --csharpVersion            C# version compatibility; allowed values: Compatible (default), Newest, NewestWithNullableReferences

  --memberAccessibility      Class and interface access level; allowed values: Public (default), Internal

  --outputType               Specifies generated classes organization; allowed values: SingleFile (default), OneClassPerFile

  --partialClasses           Mark classes as "partial"

  --classMapping             Format: {GraphQlTypeName}:{C#ClassName}; allows to define custom class names for specific GraphQL types; common reason for this is to avoid property of the same name as its parent class

  --idTypeMapping            Specifies the .NET type generated for GraphQL ID data type; allowed values: Guid (default), String, Object

  --floatTypeMapping         Specifies the .NET type generated for GraphQL Float data type; allowed values: Decimal (default), Double, Float

  --jsonPropertyAttribute    Specifies the condition for using "JsonPropertyAttribute"; allowed values: CaseInsensitive (default), CaseSensitive, Always, Never, UseDefaultAlias

  --help                     Display this help screen.

  --version                  Display version information.

```

**Comment by ZiedMayel** [comment]
I was a little confused how to execute the command. After cloning the repository I understood what you mean with.

Thank you for your explanation. It is really a great tool
________________________________
Von: Husqvik <notifications@github.com>
Gesendet: Freitag, 22. Januar 2021 16:45
An: Husqvik/GraphQlClientGenerator <GraphQlClientGenerator@noreply.github.com>
Cc: ZiedMayel <z.mayel@hotmail.com>; Author <author@noreply.github.com>
Betreff: Re: [Husqvik/GraphQlClientGenerator] Question About the First Line of the Documentation (#81)


Closed #81<https://github.com/Husqvik/GraphQlClientGenerator/issues/81>.
